test/cachingoptimizer.jl

```diff
@@ -164,10 +164,20 @@ end
 
 @testset "CachingOptimizer constructor with optimizer" begin
     s = MOIU.MockOptimizer(ModelForMock{Float64}())
-    m = MOIU.CachingOptimizer(ModelForCachingOptimizer{Float64}(), s)
+    model = ModelForCachingOptimizer{Float64}()
+    m = MOIU.CachingOptimizer(model, s)
     @test MOI.isempty(m)
     @test MOIU.state(m) == MOIU.AttachedOptimizer
     @test MOIU.mode(m) == MOIU.Automatic
+    MOI.addvariable!(s)
```

Comment:
Please split into separate tests. We should try to structure tests as:
1. Setup
2. Do something
3. `@test` that something did what we expected

This makes it easier to understand what the test is checking for and how to address it if there's a failure.

Reply:
Done :)